libraries/core_libs/consensus/include/transaction_manager/transaction_queue.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "transaction/transaction.hpp"
+
+namespace taraxa {
+
+class TransactionQueue {
+ public:
+  TransactionQueue();
+
+  bool insert(const std::shared_ptr<Transaction>& transaction);
+  bool erase(const trx_hash_t& hash);
+
+  std::shared_ptr<Transaction> get(const trx_hash_t& hash) const;
+  std::vector<std::shared_ptr<Transaction>> get(uint64_t count = 0) const;
+  bool contains(const trx_hash_t& hash) const;
+  size_t size() const;
+
+ private:
+  struct PriorityCompare {
```

Comment:
isnt it possible to use lambda instead of struct comparator ?

Reply:
I wanted to do so, but I have no idea how to do it with `this` so I can read internal structures. This is only way I make it work, but if anyone knows how to do it newer way I would appreciated it

Reply:
First step is to make people even look at this PR :)

Reply:
Everyone was tagged? what you mean?